kore/test/Test/Kore/Step/Rule/Expand.hs

```diff
@@ -0,0 +1,241 @@
+                    ( Mock.gSort0 (expandableConstructor1 x00TestSort)
+                    , makeTruePredicate
+                    )
+            actual =
+                expandOnePathSingleConstructors
+                    (metadataTools
+                        [   ( Mock.testSort0Id
+                            , noConstructor
+                                `with`
+                                    ( constructor expandableConstructor1Symbol
+                                    `with` Mock.testSort
+                                    )
+                            )
+                        ]
+                    )
+                    (Mock.fSort0 x0 `rewritesTo` Mock.gSort0 x0)
+        in assertEqualWithExplanation "" expected actual
+    , testCase "Expands variable twice." $
+        let expected =
+                Pair
+                    ( Mock.fSort0 (expandableConstructor1 Mock.a)
+                    , makeEqualsPredicate
+                        x0
+                        (expandableConstructor1 Mock.a)
+                    )
+                `rewritesTo`
+                Pair
+                    ( Mock.gSort0 (expandableConstructor1 Mock.a)
+                    , makeTruePredicate
+                    )
+            actual =
+                expandOnePathSingleConstructors
+                    (metadataTools
+                        [   ( Mock.testSort0Id
+                            , noConstructor
+                                `with`
+                                    ( constructor expandableConstructor1Symbol
+                                    `with` Mock.testSort
+                                    )
+                            )
+                        ,   ( Mock.testSortId
+                            , noConstructor `with` constructor Mock.aSymbol
+                            )
+                        ]
+                    )
+                    (Mock.fSort0 x0 `rewritesTo` Mock.gSort0 x0)
+        in assertEqualWithExplanation "" expected actual
+    ]
+  where
+    x = mkElemVar Mock.x
+    x0 = mkElemVar Mock.x0
+
+    x00TestSortVar =
+        ElementVariable
+            (Variable (testId "x0") (Just (Element 0)) Mock.testSort)
+    x00TestSort = mkElemVar x00TestSortVar
+
+    metadataTools
+        :: [(Id, Attribute.Constructors)]
+        -> SmtMetadataTools Attribute.Symbol
+    metadataTools sortAndConstructors =
+        Mock.metadataTools
+            { MetadataTools.sortConstructors = Map.fromList sortAndConstructors
+            }
+
+    expandableConstructor1Id :: Id
+    expandableConstructor1Id = testId "expandableConstructor1"
+    expandableConstructor1Symbol :: Symbol
+    expandableConstructor1Symbol =
+        Mock.symbol expandableConstructor1Id [Mock.testSort] Mock.testSort0
+        & Symbol.functional
+        & Symbol.constructor
+    expandableConstructor1
+        :: HasCallStack
+        => TermLike Variable -> TermLike Variable
+    expandableConstructor1 arg =
```

Comment:
Can we expand constructors with multiple arguments? What about multiple arguments, where only some arguments are expandable?

Reply:
Done.